application/forms/ConfigForm.php

```diff
@@ -52,9 +53,19 @@ public function setIniConfig(Config $config)
         return $this;
     }
 
-    /**
-     * {@inheritdoc}
-     */
+    public function isValid($formData)
+    {
+        $valid = parent::isValid($formData);
+
+        if ($valid && ConfigFormEventsHook::runIsValid($this) === false) {
+            $this->addErrors(ConfigFormEventsHook::getLastErrors());
```

Comment:
Please use our form notifications here. They're used everywhere else and look better.

```suggestion
            foreach (ConfigFormEventsHook::getLastErrors() as $message) {
                $this->error($message);
            }
```

Reply:
Makes sense. Done.